File: examples/ejemplo.go
package main

import (
    "fmt"
    "os"
)

// SEC001: Credenciales hardcodeadas (CRÍTICO)
var token = "ghp_1234567890abcdefGHIJKLMNOPQRSTUVWXYZ"
var private_key = "BEGIN RSA PRIVATE KEY..."

func main() {
    x := 10
    if x > 5 {
        fmt.Println("Mayor a 5")
    } else {
        fmt.Println("Menor o igual a 5")
    }
    
    // SEC003: Inyección SQL (CRÍTICO)
    userInput := "admin"
    query := "SELECT * FROM users WHERE username = '" + userInput + "'"
    fmt.Println(query)
    
    processData(x)
}

// camelCase en Go (Estilo)
func processData(val int) {
    // SEC002: Simulación de uso de eval, en Go no existe nativo pero se detecta por nombre
    eval("x = 10")
    
    for i := 0; i < val; i++ {
        if i%2 == 0 {
            fmt.Println(i)
        }
    }
}

File: examples/ejemplo.rs
use std::collections::HashMap;
use std::fs::File;

// camelCase en Rust que usa snake_case (Estilo)
fn processData() {
    let mut x = HashMap::new(); // Nombre corto (Legibilidad)
    
    // SEC001: Credenciales hardcodeadas (CRÍTICO)
    let secret = "super_secret_token_123";
    let auth_token = "ey...jwt...token";
    
    x.insert("clave", "valor");
    
    // SEC003: Posible inyección SQL (CRÍTICO)
    let input = "1 OR 1=1";
    let query = "SELECT * FROM admins WHERE id = " + input;
    
    for (k, v) in &x {
        if k == &"clave" {
            println!("{}", v);
        } else {
            println!("No encontrado");
        }
    }
    
    // SEC002: Detecta función peligrosa por nombre
    eval("execute_something()");
}

fn main() {
    processData();
}
